teams: edit members page basics

adamr2 opened the pull request.

adamr2 committed:
teams: edit members page basics

adamr2 committed:
teams: edit members page basics

adamr2 committed:
teams: Remove member functions

shakedlokits commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+
+    if (data.users.length > 0) {
+        const userId = data.users[0].id
+        console.log(`userId ${userId}`);
```
We just have to remember to remove the loggings later on 👍

wildeyes replied:
Why save the logging for later? let's remove it now

adamr2 replied:
Done

shakedlokits commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+}
+
+
+const Member = ({ member, removeMember, teamMembers, setTeamMembers }) => {
```
Nice!

shakedlokits reviewed (approved):
Very good! 👍

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+
+const getTeamMembers = gql`
+query getTeamMembers($teamId: Int!) {
+    team_members(where: {teamId: {_eq: $teamId}}) {
```
Do we have to have `_eq`? isn't it possible to just `{ teamId: $teamId }`?

adamr2 replied:
I don't think we can, double check me [here](https://hasura.io/docs/1.0/graphql/manual/queries/query-filters.html#equality-operators-eq-neq)

wildeyes replied:
fuck it

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+import React, { ChangeEvent, useEffect, useState } from 'react'
```
Could you please prettier-fy the file? there's already a .prettierrc, you use vs-code so just a single `prettier` extension :)

adamr2 replied:
Done

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+
+
+const deleteTeamMember = gql`
+mutation DeleteTeamMembers($uid: Int!, $tid: Int!) {
```
Please use full names, there's no need for obscurity ;)

wildeyes replied:
uid, tid

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+    }, [])
+
+    useEffect(() => {
+        console.log("2 called")
```
Please remove left over console logs

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+                alert(errors[0])
+                throw errors[0]
+            }
+            return 'success'
```
Just a note, if you want to use a non false value returned to check whether addMember succeded or not, I'd expect to return the object of the new user, but that's up to you :)

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+
+const removeMember = async (memberData) => {
+    // remove user from team
+    console.log('removing user')
```
Please remove left over console logs

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+        mutation: deleteTeamMember,
+        variables: { uid: memberData.user.id, tid: memberData.teamId }
+    })
+    console.log(data)
```
here too please

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+
+    let error;
+    useEffect(() => {
+        console.log("loaded")
```
also

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+
+
+    const router = useRouter();
+    const [teamId, setTeamId] = useState((typeof router.query.id === 'string') ? router.query.id : router.query.id[0]);
```
Please document why this is needed

wildeyes replied:
the whole `typeof router.query.id === 'string') ? router.query.id : router.query.id[0]`

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+
+        const userMail: string = formData.get('email').toString()
+        const userId = await addMember(teamId, userMail)
+        if (userId) {
```
Why do we router.replace again?

wildeyes commented on services/authentication/resolvers.ts:
```diff
@@ -9,7 +9,7 @@ export const jwtAlgorithm = 'HS512'
     host: 'postgres',
     user: 'postgres',
-    password: 'changeme',
+    password: 'password',
```
nice ;)

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+            <h1>Edit Members Screen</h1>
+            <h3>Team {teamId}</h3>
+            {teamMembers.map((member, i) => (
+                <Member key={i} member={member} teamMembers={teamMembers} setTeamMembers={setTeamMembers} removeMember={removeMember} />
```
@shakedlokits if the functions exist in outer scope, should one pass them into props? i would recomend against, what do you think?

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+        const formElement = document.forms[0]
+        const formData = new FormData(formElement)
+
+        const userMail: string = formData.get('email').toString()
```
is there a need to cast to string if you do a .toString()?

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+    const router = useRouter();
+    const [teamId, setTeamId] = useState((typeof router.query.id === 'string') ? router.query.id : router.query.id[0]);
+    const [data, setData] = useState(null)
+    const [teamMembers, setTeamMembers] = useState([]);
```
Would you like to try and imitate `useQuery` from `TeamProfilePage`? it would supposedly know when you changed the page and reload the data accordingly

adamr2 replied:
I tried,
it caused problems with useEffect.
I think it had something to do with next.js pre-rendering but I'm not sure..
The problem was that the app failed after I refreshed due to a "change in the number of hooks"
I'll keep it this way and maybe I'll try to use useQuery later down the road

wildeyes commented on providers/teams/pages/members/[id].tsx:
```diff
@@ -0,0 +1,215 @@
+    console.log(data.delete_team_members.returning);
+    if (errors) {
+        alert(errors[0])
+        throw errors[0]
```
@shakedlokits we need to form a plan on what to do with all these errors, WDYT?

wildeyes reviewed (commented):
Fantastic work, left some notes, and then some :)
Kol Hakavod

adamr2 committed:
teams: tidy up
resolves https://github.com/DataHackIL/ninjathon/pull/55#pullrequestreview-395013922

adamr2 committed:
teams: manipulate team challenges

adamr2 committed:
teams: multiple features:
* challenge functions
* useQuery hook
* refetch cached queries in mutations
* Better re-render after updates

resolves: https://github.com/DataHackIL/ninjathon/pull/55#discussion_r409868185
